### PR TITLE
Add a ttfb graph to the dss dashboard.

### PR DESCRIPTION
**Pull request by DailyDreaming**

### CODE REVIEW

File: terraform/modules/env-dashboards/dss-dashboard.tf [hunk 1]
```diff
@@ -1,26 +1,133 @@
...
+                "show": true,
+                "values": []
+            },
+            "yaxes": [
+                {
+                    "format": "short",
+                    "label": null,
+                    "logBase": 1,
+                    "max": null,
+                    "min": null,
```
Comment (chmreid) [inline terraform/modules/env-dashboards/dss-dashboard.tf:114]: Would a y-min of 0 make sense for this graph?

**Review by chmreid — APPROVED** [review]
One nit about the y-axis minimum, other than that LGTM

File: terraform/envs/prod/env-dashboards/variables.tf [hunk 2]
```diff
@@ -5,6 +5,7 @@ locals {
 
   aws_cloudwatch_data_source_name = "account-cloudwatch"
   aws_logs_data_source_name       = "${data.terraform_remote_state.account-dashboards.aws_logs_data_source_name}"
+  hca_logs_ttfb_datasource_name   = "hca-logs-ttfb"
```
Comment (amarjandu) [inline terraform/envs/prod/env-dashboards/variables.tf:8]: should the prod source_name have an indication to the account?

File: terraform/modules/env-dashboards/datasources.tf [hunk 3]
```diff
@@ -11,9 +11,25 @@ data "external" "json" {
...
+
+  hca_logs_ttfb_datasource = <<EOF
+{
+  "name": "${local.hca_logs_ttfb_datasource_name}",
+  "type": "elasticsearch",
+  "database": "[ttfb-]YYYY-MM-DD",
+  "url": "http://0.0.0.0:9200",
+  "access": "proxy",
+  "jsonData": {
+    "interval": "Daily",
```
Comment (amarjandu) [inline terraform/modules/env-dashboards/datasources.tf:27]: is this the rate for how fast the TTFB is updated? can this be hourly?

File: terraform/modules/env-dashboards/dss-dashboard.tf [hunk 4]
```diff
@@ -1,26 +1,133 @@
...
+            "links": [],
+            "panels": [{
+            "aliasColors": {},
+            "bars": true,
+            "dashLength": 10,
+            "dashes": false,
+            "datasource": "${local.hca_logs_ttfb_datasource_name}",
+            "fill": 1,
+            "fillGradient": 0,
+            "gridPos": {
```
Comment (amarjandu) [inline terraform/modules/env-dashboards/dss-dashboard.tf:31]: you will need to ensure that this gridPos does not collide with others in the file, if you have inserted this into the file i believe you have to modify all the other grid pos for the other panels due to them shifting down. an alternative is to place the new panel at the bottom so you dont have to modify all the other gridpos.

File: terraform/modules/env-dashboards/dss-dashboard.tf [hunk 5]
```diff
@@ -1,26 +1,133 @@
...
+                "h": 8,
+                "w": 24,
+                "x": 0,
+                "y": 8
+            },
+            "id": 22,
```
Comment (amarjandu) [inline terraform/modules/env-dashboards/dss-dashboard.tf:37]: verify this does not collide with other panel IDs in the file, or you get weird bugs.

File: terraform/modules/env-dashboards/dss-dashboard.tf [hunk 6]
```diff
@@ -1,26 +1,133 @@
...
+            "legend": {
```
Comment (amarjandu) [inline terraform/modules/env-dashboards/dss-dashboard.tf:38]: please consider adding a legend and an average value within this legend for easier observation. the bucket panels have this if you need an example.

File: terraform/modules/env-dashboards/dss-dashboard.tf (hunk 1 above)
Comment (amarjandu) [reply terraform/modules/env-dashboards/dss-dashboard.tf]: +1 the y-min is from -1 to +1 which is a little strange.

File: terraform/modules/env-dashboards/dss-dashboard.tf [hunk 7]
```diff
@@ -1,26 +1,133 @@
...
+            "timeFrom": null,
+            "timeRegions": [],
+            "timeShift": null,
+            "title": "GET /file Latency (Time to First Byte)",
+            "tooltip": {
+                "shared": true,
+                "sort": 0,
+                "value_type": "individual"
+            },
+            "transparent": true,
```
Comment (amarjandu) [inline terraform/modules/env-dashboards/dss-dashboard.tf:99]: Other panels are not transparent, please set this to `false`

**Review by amarjandu — CHANGES_REQUESTED** [review]
There is some tuning to the panel that needs to be done; please take a look below. I also dont see any data being input into the graph. If there is an issue with the data source we can work out another way to get the data into cloudwatch metrics. 

Once data is visible we might want to take this from a bar graph into a line graph. 

Screen Shot 2019-12-06 at 2 14 09 PM

File: terraform/modules/env-dashboards/datasources.tf (hunk 3 above)
Comment (DailyDreaming) [reply terraform/modules/env-dashboards/datasources.tf:27]: It's not.  Daily is appropriate.  Please see documentation.

File: terraform/modules/env-dashboards/dss-dashboard.tf (hunk 4 above)
Comment (DailyDreaming) [reply terraform/modules/env-dashboards/dss-dashboard.tf:31]: Yep.

File: terraform/modules/env-dashboards/dss-dashboard.tf (hunk 5 above)
Comment (DailyDreaming) [reply terraform/modules/env-dashboards/dss-dashboard.tf:37]: Yep.

File: terraform/modules/env-dashboards/dss-dashboard.tf (hunk 6 above)
Comment (DailyDreaming) [reply terraform/modules/env-dashboards/dss-dashboard.tf:38]: I think it's cleaner as is.

File: terraform/modules/env-dashboards/dss-dashboard.tf (hunk 7 above)
Comment (DailyDreaming) [reply terraform/modules/env-dashboards/dss-dashboard.tf:99]: Eh, you're probably right, adding a new style is confusing.

**Comment by DailyDreaming** [comment]
This was still open when I left for other projects.  Just had to address some comments before closing.

**Comment by amarjandu** [comment]
@DailyDreaming  can this issue be closed?

**Comment by DailyDreaming** [comment]
@amarjandu Ha, indeed, I think I beat your comment response notification.  ;D

**Pull request closed without merging**